executor/include/executor/default/omp_executor.hpp

```diff
@@ -36,6 +36,16 @@ struct omp_executor {
     int idx;
   };
 
+  std::size_t max_threads;
+
+  omp_executor() {
+    this->max_threads = omp_get_max_threads();
```

Comment:
Why not `omp_executor(): max_threads(omp_get_max_threads()) {}`?

Reply:
`omp_get_max_threads` would not be available on non-omp devices.

Reply:
Ideally, you should delegate to the other ctor in this.

The other ctor should use initializer list instead of in-body assignment

Reply:
```cpp
omp_executor(): omp_executor(0) {}
```
Keep macros in one place